Realm/Realm/Native/Configuration.cs

```diff
@@ -50,6 +50,9 @@ internal string Path
         [MarshalAs(UnmanagedType.U1)]
         internal bool in_memory;
 
+        [MarshalAs(UnmanagedType.U1)]
+        internal bool additive_discovered;
```

Comment:
This is not used for non-sync configurations, so we should probably move it to the sync config. Also, let's replace it with a SchemaMode enum instead.

Reply:
I'm in doubt on a thing: should the enum in C# have only the 2 additive options or should it reflect the whole enum that's in the object store? There are pros and cons to both ways, so I can't decide.

Reply:
Let's only have the two options - if we need more, we can add more - it's an internal API, so we can change it however we please.